src/roboat_perception/roboat_perception/radio.py

```diff
@@ -7,49 +7,64 @@
 from std_msgs.msg import String
 import subprocess
 import os
+from .data.dtp import create_packet, decode_packet
+from .data.sliplib import Driver
+from bitstring import Bits, BitArray
 
 class Radio(Node):
     def __init__(self):
         super().__init__('radio')
+        self.slip_driver=Driver() #sliplib
 
         self.publisher_ = self.create_publisher(String, 'recv', 10)
-        self.subscription = self.create_subscription(String, 'send', self.listener_callback, 10)
-        
-        timer_period = 30
-        self.timer = self.create_timer(timer_period, self.publish) # Temp Timer
+        self.subscription = self.create_subscription(String, 'send', self.write_to_radio, 10)
 
-        M0 = 0
-        M1 = 2
-        AUX = 3
+        self.M0 = 17 #eletrical pin numbers
+        self.M1 = 27
+        self.AUX = 22
 
         GPIO.setmode(GPIO.BCM)
-        GPIO.setup(M0, GPIO.OUT)
-        GPIO.setup(M1, GPIO.OUT)
-        GPIO.setup(AUX, GPIO.IN)
-        self.radio_ser = serial.Serial('/dev/ttyS0', baudrate=9600, timeout=2)
+        GPIO.setup(self.M0, GPIO.OUT)
+        GPIO.setup(self.M1, GPIO.OUT)
+        GPIO.setup(self.AUX, GPIO.IN)
+        self.radio_ser = serial.Serial('/dev/ttyS0', baudrate=9600, timeout=0)
 
         self.self_test()
         self.get_logger().info("Radio Initialized")
-        self.set_power_saving_mode()
+        self.set_normal_mode()
+
+        timer_period_sec = 1
+        self.timer = self.create_timer(timer_period_sec, self.read_from_radio) # Temperary Timer
     
-    def publish(self):
-        msg = self.radio_ser.read(size=10)
-        if len(msg>2):
-            sendMSG.data = msg.decode('utf-8')
-            self.publisher_.publish(msg)
-            self.get_logger().info("Message Sent")
+    def read_from_radio(self): # Needs to be tested
+        self.get_logger().info("Started Reading")
+        if GPIO.input(self.AUX) == GPIO.LOW:
+            self.block_until_radio_ready()
```

Comment:
These lines aren't necessary, since the linux buffer will ensure we get all recvs.